src/components/ContractName.tsx

```diff
@@ -2,19 +2,35 @@ import { Suspense, memo } from 'react'
 import { useSnapshot } from 'valtio'
 import { wordBreak } from 'classnames/tailwind'
 import ContractNamesStore from 'stores/ContractNamesStore'
+import SealCredStore from 'stores/SealCredStore'
 import truncateMiddleIfNeeded from 'helpers/truncateMiddleIfNeeded'
 
 const addressText = wordBreak('break-all')
 
 interface ContractNameProps {
   address: string
   truncate?: boolean
+  clearType?: boolean
 }
 
-function ContractNameSuspended({ address, truncate }: ContractNameProps) {
+function ContractNameSuspended({
+  address,
+  truncate,
+  clearType,
+}: ContractNameProps) {
+  const { emailDerivativeContracts = [], ERC721derivativeContracts = [] } =
+    useSnapshot(SealCredStore)
   const { contractNames } = useSnapshot(ContractNamesStore)
-  const contractName = contractNames[address]
-  if (!contractNames[address]) ContractNamesStore.fetchContractName(address)
+  let contractName = contractNames[address]
+  if (!contractName) ContractNamesStore.fetchContractName(address)
```

Comment:
What if there's no `contractName`, then it tries to fetch it, no `contractName` again, you'll get an infinite loop here?

Reply:
No, the `contractName` will be requested only once)